package-res/resources/web/pentaho/visual/ccc/abstract/View.js

```diff
@@ -1382,21 +1382,58 @@ define([
       }
     },
 
-    _renderCore: function() {
-      var domContainer = this.domContainer;
-      while(domContainer.firstChild) {
-        domContainer.removeChild(domContainer.firstChild);
-      }
+    /**
+     * Creates a chart of a given class and with given options.
+     *
+     * @param {!Class.<pvc.BaseChart>} ChartClass The CCC chart constructor.
+     * @param {!object} options The CCC options.
+     * @return {!pvc.BaseChart} The created chart instance.
+     * @protected
+     */
+    _createChart: function(ChartClass, options) {
+      return new ChartClass(options);
+    },
+
+    /**
+     * Gets the current value of maximum number of visual elements.
+     *
+     * @return {number} The maximum number when defined and greater than `0`; `-1`, otherwise.
+     */
+    _getVisualElementsCountMax: function() {
 
-      var ChartClass = pvc[this._cccClass];
+      // TODO: Currently, this is DET dependent...
 
-      this._chart = new ChartClass(this.options);
+      var modelTypeApp = this.model.type.application;
+      var cellCountMax = +(modelTypeApp && modelTypeApp.optimalMaxDataSize);
```

Comment:
Ok. But then maybe we should increase the DET limit. I think 5k was chosen thinking about multiple measures, maybe 10k would be ok?